app/contact/dns.py

```diff
@@ -0,0 +1,435 @@
+        :return: DNS RR answer or None if rrtype is not accounted for
+        :rtype: dnslib.RR or None
+        """
+        if rrtype == "TXT":
+            return RR(rname=qname, rtype=QTYPE.TXT, rclass=CLASS.IN, ttl=0, rdata=TXT(data))
+        elif rrtype == "A":
+            return RR(rname=qname, rtype=QTYPE.A, rclass=CLASS.IN, ttl=1, rdata=A(data))
+        else:
+            return None
+
+    def _generate_response(self, request, rrs):
+        """
+        Generates a reply to the request with any input RR answers.
+
+        :param request: DNSRecord request
+        :param rrs: Array of RR answers
+        :return: DNSRecord DNS reply
+        """
+        resp = request.reply()
+        if isinstance(rrs, list):
+            for rr in rrs:
+                resp.add_answer(rr)
+        else:
+            resp.add_answer(rrs)
+        return resp
+
+    @staticmethod
+    def decode_bytes(s):
+        """
+        Receives a zlib compressed base64 string and returns the plaintext.
+
+        :param s: Zlib compressed, base64 encoded string
+        :type s: str
+        :return: plaintext
+        :rtype: str
+        """
+        return str(zlib.decompress(urlsafe_b64decode(s)).decode().replace('\n', ''))
+
+    @staticmethod
+    def decode_raw(s):
+        """
+        Receives a zlib compressed base64 string and returns raw bytes
+
+        """
+        return zlib.decompress(urlsafe_b64decode(s))
+
+    @staticmethod
+    def encode_string(s):
+        """
+        Receives a plaintext string, zlib compresses it, and encodes it into a base64 string.
+
+        :param s: plaintext string
+        :type s: str
+        :return: Zlib compressed, base64 encoded string
+        :rtype: str
+        """
+        if isinstance(s, str):
+            s = s.encode()
+        return str(urlsafe_b64encode(zlib.compress(s, 9)).decode())
+
+    @staticmethod
+    def chunk_string(s, n=150):
+        """
+        Takes a string and splits it into x chunks of n length. The default is 150 to safely account for DNS overhead.
+
+        :param s: string
+        :type s: str
+        :param n: Number of characters per chunk
+        :type n: int
+        :return: Array of n-length chunks of original string
+        :rtype: list
+        """
+        return [s[i:i+n] for i in range(0, len(s), n)]
+
+    @staticmethod
+    def chunk_data_for_packets(data, chunk_size):
+        """
+        Takes a list of string chunks and creates a list of `chunk_size` lists from the original data.
+
+        :param data: list of string chunks
+        :type data: list
+        :param chunk_size: number of string chunks per data chunk
+        :type chunk_size: int
+        :return: List of `chunk_size` lists from original string chunk list.
+        """
+        ret = []
+        while len(data) > 0:
+            ret.append(data[:chunk_size])
+            del data[:chunk_size]
+        return ret
+
+
+class DNS(C2Passive):
+    """
```

Comment:
Move this out of the Private function section.  I think this should be first class in the file...